raster/r.series/main.c

```diff
@@ -397,89 +454,158 @@ int main(int argc, char *argv[])
         out->quantile = (parm.quantile->answer && parm.quantile->answers[i])
                             ? atof(parm.quantile->answers[i])
                             : 0;
-        out->buf = Rast_allocate_d_buf();
+        out->buf = G_calloc((size_t) bufrows * ncols, sizeof(DCELL));
         if (menu[method].outtype == -1)
             out->fd = Rast_open_new(output_name, intype);
         else
             out->fd = Rast_open_new(output_name, menu[method].outtype);
     }
 
     /* initialise variables */
-    values = G_malloc(num_inputs * sizeof(DCELL));
-    values_tmp = G_malloc(num_inputs * sizeof(DCELL));
+    values = G_malloc(nprocs * sizeof *values);
+    values_tmp = G_malloc(nprocs * sizeof *values_tmp);
+    for (t = 0; t < nprocs; t++) {
+        values[t] = G_malloc(sizeof(DCELL) * num_inputs);
+        values_tmp[t] = G_malloc(sizeof(DCELL) * num_inputs);
+    }
     values_w = NULL;
     values_w_tmp = NULL;
     if (have_weights) {
-        values_w = (DCELL(*)[2]) G_malloc(num_inputs * 2 * sizeof(DCELL));
-        values_w_tmp = (DCELL(*)[2]) G_malloc(num_inputs * 2 * sizeof(DCELL));
+        values_w = G_malloc(nprocs * sizeof *values_w);
+        values_w_tmp = G_malloc(nprocs * sizeof *values_w_tmp);
+        for (t = 0; t < nprocs; t++) {
+            values_w[t] = (DCELL(*)[2]) G_malloc(sizeof(DCELL) * num_inputs * 2);
+            values_w_tmp[t] = (DCELL(*)[2]) G_malloc(sizeof(DCELL) * num_inputs * 2);
+        }
     }
 
-    nrows = Rast_window_rows();
-    ncols = Rast_window_cols();
-
     /* process the data */
     G_verbose_message(_("Percent complete..."));
 
-    for (row = 0; row < nrows; row++) {
-        G_percent(row, nrows, 2);
+    int computed = 0;
+    int written = 0;
 
```

Comment:
`row = 0;` to avoid
```
main.c:600:5: warning: ‘row’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     G_percent(row, nrows, 2);
```

Reply:
yeap fixed this as well. Supposed to be G_percent(nrows, nrows, 2);